### PR TITLE
feat: opsRule supports workspace config

### PR DESCRIPTION
**Pull request by SparkYuan**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
feat: opsRule supports workspace config
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/modules/generators/trait/ops_rule_generator.go [hunk 1]
```diff
@@ -1,36 +1,54 @@
...
 	app *appmodule.AppConfiguration,
+	ws *workspaceapi.Workspace,
 ) (modules.Generator, error) {
+	if project == nil || len(project.Name) == 0 {
+		return nil, fmt.Errorf("project name must not be empty")
+	}
+	if ws == nil {
+		return nil, fmt.Errorf("workspace must not be empty")
+	}
+	wsConfig, err := workspace.GetProjectModuleConfigs(ws.Modules, project.Name)
```
Comment (healthjyk) [inline pkg/modules/generators/trait/ops_rule_generator.go:41]: You can get moduleConfigs in `pkg/modules/generators/app_configurations_generator.go`, and only use the opsRule config in workspace as input.
[PR#709](https://github.com/KusionStack/kusion/pull/709) does so.

File: pkg/modules/generators/trait/ops_rule_generator.go [hunk 2]
```diff
@@ -39,14 +57,16 @@ func NewOpsRuleGeneratorFunc(
...
 	return func() (modules.Generator, error) {
-		return NewOpsRuleGenerator(project, stack, appName, app)
+		return NewOpsRuleGenerator(project, stack, appName, app, ws)
 	}
 }
 
 func (g *opsRuleGenerator) Generate(spec *intent.Intent) error {
-	if g.app.OpsRule == nil {
+	// opsRule does not exist in AppConfig and workspace config
+	if g.app.OpsRule == nil && g.workspaceConfig["opsRule"] == nil {
```
Comment (healthjyk) [inline pkg/modules/generators/trait/ops_rule_generator.go:69]: make "opsRule" a const is better.

File: pkg/modules/inputs/trait/ops_rule.go [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+			return intstr.IntOrString{}, nil
+		}
+		var wsValue string
+		wsValue, isString := workspaceConfig[OpsRuleConst][MaxUnavailableConst].(string)
+		if !isString {
+			temp, isInt := workspaceConfig[OpsRuleConst][MaxUnavailableConst].(int)
+			if isInt {
+				wsValue = strconv.Itoa(temp)
+			} else {
+				return intstr.IntOrString{}, errors.New("illegal workspace config. opsRule.maxUnavailable in the workspace config is not string or int")
```
Comment (healthjyk) [inline pkg/modules/inputs/trait/ops_rule.go:39]: there is space in `config. opsRule.maxUnavailable `

Comment (SparkYuan) [reply pkg/modules/inputs/trait/ops_rule.go:39]: They are two sentences. I add a space deliberately...

**Review by healthjyk — APPROVED** [review]

**Pull request merged**